### DISCUSSION TITLE
Atom mapping before and after removing hydrogens

### DISCUSSION THREAD

**Discussion by zarkoivkovicc** [rdkit/rdkit#7909] [Q&A] [answered]
I want to get the atom mapping before and after removing hydrogens in a molecule. The reason for this is that I have some atomic properties for an rdkit molecule (not canonical) but I want to show them in the molecule without hydrogen atoms. I can't find a way to map the atom numbers before and after removing hydrogens. For before and after canonization process there is _smilesAtomOutputOrder property, but there isn't a similar property for removing hydrogens.

**Comment by greglandrum** [comment, marked as answer]
@zarkoivkovicc you can find the mapping by doing a substructure match between the H-stripped molecule and the original:
```
In [12]: ps = Chem.SmilesParserParams()

In [13]: ps.removeHs = False

In [14]: m = Chem.MolFromSmiles('[H]OC(O[H])CC',ps)

In [15]: m2 = Chem.RemoveHs(m)

In [16]: m.GetSubstructMatch(m2)
Out[16]: (1, 2, 3, 5, 6)

In [17]: mv = m.GetSubstructMatch(m2)

In [18]: for i,j in enumerate(mv):
    ...:     print(f'{j}->{i}')
    ...: 
1->0
2->1
3->2
5->3
6->4
```

**Reply by zarkoivkovicc** [reply]
Thank you very much for the fast reply!

**Reply by greglandrum** [reply]
I'm reopening this so that it's still visible to others in case the question comes up again
